intermediate_source/TCPStore_libuv_backend.rst

```diff
@@ -0,0 +1,100 @@
+Introduction to Libuv TCPStore Backend
+======================================
+**Authors**: `Xilun Wu <https://github.com/XilunWu>`_
+
+.. note::
+    |edit| View and edit this tutorial in `github <https://github.com/pytorch/tutorials/blob/main/intermediate_source/TCPStore_libuv_backend.rst>`__.
+
+Prerequisites:
+
+- `TCPStore API Documents <https://pytorch.org/docs/main/distributed.html#torch.distributed.TCPStore>`__
+
+Introduction
+------------
+
+Recently, we have rolled out a new TCPStore server backend using libuv, a third-party library for asynchronous I/O. The motivation of this new server backend
+is to address the scalability and robustness challenge in large-scale jobs, for example, distributed training with more than 1024 ranks. In PyTorch 2.4, we
+decide to make this new backend as the default for TCPStore server and this tutorial will cover the impact of this change on user code and how to stay with
+using the old backend.
+
+Impact
+------
+
+TCPStore currently does not support initialization with an listen fd when using libuv backend. If user wants to keep using this initialization method, user can
+simply pass ``use_libuv=False`` to stay with the old TCPStore backend.
+
+.. code:: python
+
+    import socket
+
+    import torch
+    import torch.distributed as dist
+
+    listen_sock: socket.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    listen_sock.bind(("localhost", 0))
+    addr, port, *_ = listen_sock.getsockname()
+    listen_fd = listen_sock.detach()
+
+    tcpstore = dist.TCPStore(addr, port, 1, True, master_listen_fd=listen_fd)  # expect NotImplementedError
+    tcpstore = dist.TCPStore(addr, port, 1, True, master_listen_fd=listen_fd, use_libuv=False)  # OK. Use old backend
+
+
+Exit Route 1: Pass ``use_libuv=False`` to TCPStore Initialization
+-----------------------------------------------------------------
+
+As the above code snippet shows, if user calls TCPStore init method to create a store, simply passing ``use_libuv=False`` allows user to remain using the old
+TCPStore backend. This override has the highest priority over other approaches determining which backend the TCPStore server should choose.
+
+
+Exit Route 2: Add ``use_libuv=0`` to ``init_method`` at ProcessGroup Initialization
+-----------------------------------------------------------------------------------
+
+ProcessGroup creates a TCPStore if user does not explicitly pass one to its initialization. User can add the query option ``use_libuv=0`` to ``init_method`` when
+initializing the ProcessGroup. This approach has lower priority than Exit Route 1.
```

Comment:
not sure i understand what is meant by priority here.

Reply:
I was meaning that the first approach can override the second, and so on. Like, if user sets env var `USE_LIBUV=1` and also pass `use_libuv=0` in `init_method`, the old store backend will be chosen.